### PR TITLE
CLOUDP-155908: automation for generating sdk

### PR DESCRIPTION
**Pull request by wtrocki**
Automation for generating SDK

## Review

This PR brings automation to fetch generate and automatically create PRs with SDK changes:
https://github.com/mongodb/go-client-mongodb-atlas/pull/359

PR is created as github-actions bot - we might use our own one if needed.

# Notes

I will handle pings for the team in separate PR. 
Base branch is automatically derrived from the trigger branch. This will enable us to run updates on v2 at later phase.

### CODE REVIEW

File: v2/Makefile [hunk 1]
```diff
@@ -5,18 +5,25 @@ example:
...
-generate:
+.PHONY: generate_client
+generate_client:
 	(cd ./tools && \
 	OPENAPI_FOLDER="../openapi" SDK_ROOT="../" \
 	./scripts/generate.sh)
 
+.PHONY: compile_client
+compile_client:
+	go build ./examples/example_cluster_aws.go
```
Comment (colm-quinn) [inline v2/Makefile:24]: [q] Would this list need to keep growing over time as more examples are added? Can we do better and build a root example file that contains everything needed for the SDK startup target?

Comment (colm-quinn) [reply v2/Makefile:24]: Possible naming - example is actually the client entry point and not the best naming

File: .github/workflows/autoupdate.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+name: Generate Client
+on:
+ ## For verification only
+ push:
+   branches:
+     - CLOUDP-155908
```
Comment (colm-quinn) [inline .github/workflows/autoupdate.yaml:6]: Do you mean to commit the CLOUDP as part of the automation? Seems the branch should be v2 or similar. If this is a prerelease branch, maybe use that name instead?

I'd expect after this commit, you can use the automation to publish an SDK, even if it's a private branch.

**Review by colm-quinn — COMMENTED** [review]
Some comments on style and naming - great to see automation starting so soon!

File: .github/workflows/autoupdate.yaml (hunk 2 above)
Comment (wtrocki) [reply .github/workflows/autoupdate.yaml:6]:
```suggestion
 push:
   branches:
     - v2
```

Comment (wtrocki) [reply .github/workflows/autoupdate.yaml:6]: Made change to verify automation. Not needed anymore.

File: .github/workflows/autoupdate.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        if: steps.verify-changed-files.outputs.files_changed == 'true'
+        working-directory: ./v2
+        run: make verify_client
+      - uses: peter-evans/create-pull-request@v4
+        if: steps.verify-changed-files.outputs.files_changed == 'true'
+        with:
+          title: "OpenAPI Client Update"
+          commit-message: "OpenAPI Client Update"
+          branch: client
+          branch-suffix: timestamp
```
Comment (wtrocki) [inline .github/workflows/autoupdate.yaml:39]: I think we can remove this strategy and push it to existing PRs. 
Multiple PRs with SDKs will not be desired.

Comment (wtrocki) [reply .github/workflows/autoupdate.yaml:39]:
```suggestion
```
See: https://github.com/peter-evans/create-pull-request#alternative-strategy---always-create-a-new-pull-request-branch

File: .github/workflows/autoupdate.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        working-directory: ./v2
+        run: make verify_client
+      - uses: peter-evans/create-pull-request@v4
+        if: steps.verify-changed-files.outputs.files_changed == 'true'
+        with:
+          title: "OpenAPI Client Update"
+          commit-message: "OpenAPI Client Update"
+          branch: client
+          body: |
+            Automatic update for MongoDBAtlas Go Client
```
Comment (wtrocki) [inline .github/workflows/autoupdate.yaml:40]:
```suggestion
            Automatic update for MongoDB Atlas Go Client.
            PR contains autogenerated changes for the 
            MongoDB Atlas client.
            
            ## Review procedure
            
            1. Compare changes in the OpenAPI file
            2. Review if there was a new platform version introduced 
            3. Review client changes. 
```

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**